### ISSUE TITLE
Improve CTS for attached case

### ISSUE THREAD

**Issue by oharboe** [The-OpenROAD-Project/OpenROAD#4666]
Which seems to have resulted in a strange CTS result, look at the macro leaf nodes:

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/a1963a0c-9476-4e85-9389-ccb18c1da196)

Unbalanced at the leaf buffer:

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/fa8d2825-2850-4898-925c-14aa5caf6628)

_Originally posted by @oharboe in https://github.com/The-OpenROAD-Project/OpenROAD/discussions/4660#discussioncomment-8453920_

**Comment by oharboe** [comment]
To view the result:

```
$ . vars-mock-array-asap7-base.sh
$ ODB_FILE=results/asap7/mock-array/base/4_1_cts.odb openroad -gui scripts/gui.tcl
```

The problem goes away if I revert https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/1799:

```
$ . vars-mock-array-asap7-base.sh
$ export CTS_ARGS="-insertion_delay -sink_clustering_enable -balance_levels -distance_between_buffers 60"
$ openroad -no_init ${SCRIPTS_DIR}/cts.tcl
```

Results in the expected CTS, 4 macros hanging off each buffer and not 4 macros + a buffer hanging off a buffer now seen on master:


![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/166fa008-9e3c-4c31-b15e-77e65757d821)

**Comment by oharboe** [comment]
@maliberty FYI, this seems to be the reason we see some regressions after the cleanup in https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/1799

**Comment by maliberty** [comment]
I don't think there were any regression related to that PR.  mock array is unchanged by that PR.

**Comment by oharboe** [comment]
> I don't think there were any regression related to that PR. mock array is unchanged by that PR.

Yes, but that is only because mock-array overrides CTS_ARGS completely. I only picked up the `-distance_between_buffers 60` because asap7 had it by default. My *intention* was to use the default arguments and only add `-insertion_delay`.

So... I can complete the cleanup of https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/1799 in https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/1802 when the feature request in this issue is completed.

**Comment by maliberty** [comment]
If you mean we would have seen some regression with a different PR than was actually merged then I agree.

**Comment by oharboe** [comment]
> If you mean we would have seen some regression with a different PR than was actually merged then I agree.

I think the sequencing of PRs and feature requests worked out well, even if it was not intended :-)

Otherwise everything would have stalled pending this feature request having been completed.

**Comment by maliberty** [comment]
macro trees have had substantial improvements.  Reopen if still reproducible
